horizomer/utils/tree.py

```diff
@@ -91,6 +144,66 @@ def intersect_trees(tree1, tree2):
     return (tree1_lap, tree2_lap)
 
 
+def collapse_short_branches(tree, cutoff):
+    """Collapse internal nodes with branch length below cutoff.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+        tree to search for nodes to collapse
+    cutoff : int or float
+        branch length cutoff under which node should be collapsed
+
+    Returns
+    -------
+    skbio.TreeNode
+        resulting tree with short branches collapsed
+
+    Notes
+    -------
+    Nodes without branch length are considered as having zero branch length and
+    will be collapsed.
+    """
+    tcopy = tree.copy()
+    nodes_to_collapse = []
+    for node in tcopy.non_tips():
```

Comment:
why do you exclude tips? Don't they also have lengths, which might be so small that you need to "merge" them with parent?

Reply:
Tips have lengths, however they won't intefer with the topology. If we collapse tips they basically become zero branch length. But the topology remains.

Reply:
let's assume a cutoff of 1.0
![qiyun](https://user-images.githubusercontent.com/11960616/35580333-9678c5f8-059d-11e8-943e-bcea43dcf37c.png)

Reply:
Hi @sjanssen2 I tested your case. Result suggests that there is no issue without recursion. See here:
![screen shot 2018-02-02 at 1 38 49 pm](https://user-images.githubusercontent.com/4396343/35756035-852c2e42-081e-11e8-92c5-1ab8390b8778.png)